aimmo-game/simulation/worker_manager.py

```diff
@@ -18,14 +18,12 @@ class _WorkerManagerData(object):
     """
     This class is thread safe
     """
-
     def __init__(self, game_state, user_codes):
         self._game_state = game_state
         self._user_codes = user_codes
         self._lock = Semaphore()
 
     def _remove_avatar(self, user_id):
-        assert self._lock.locked
         self._game_state.remove_avatar(user_id)
         del self._user_codes[user_id]
```

Comment:
Well the assert doesn't prevent that from happening either - reasoning for removing it was that it's a private method that only gets called inside a `with self._lock` block in other methods. Better than an assert would be to change `_lock` to a re-entrant lock and acquire it again.